torch/csrc/jit/mobile/backport_manager.cpp

```diff
@@ -327,28 +227,162 @@ std::stringstream backport_v6_to_v5(std::stringstream& input_model_stream) {
       /*archive_name=*/"constants",
       /*archive_dir=*/"",
       /*tensor_dir=*/"constants/",
       /*use_storage_context=*/true,
       storage_context);
-  writeArchiveV5(
+  write_archive_v5_to_v7(
       writer_bytecode,
       bytecode_tuple,
       /*archive_name=*/"bytecode",
       /*archive_dir=*/"",
       /*tensor_dir=*/"constants/",
       /*use_storage_context=*/true,
       storage_context);
+  return ouput_model_stream;
+}
+
+} // namespace
+
+/*
+ To add next backport function, for example, backport_vn_to_vn-1, create an
+ anonymous namespace with a backport_vn_to_vn-1 function + other necessary
+ customized function. If a function can be reused by other backport functions,
+ move it to the utility function group. It will be easier to split out
+ backport_manager.cpp to smaller files when it grows too long.
+
+ How to add backport_v{i}_to_v{i-1} ?
+ There are two options:
+ 1) [Format change only, recommended] Constrcut a reader with the
+ input_model_stream, modify the file, and use PyTorchWriter to write it to
+ output_model_stream. See backport_v5_to_v4.
+
+ 2) [Both format and content change] ]Use torch.jit.load() to load the stream,
+ and save it to output_model_stream.
+
+ The first option is preferred, because it will be purely format change, and
+ the model doesn't need to go through inline again and model content will
+ remain the same.
+
+ A note for manipulate stringstream, it's recommend to declare a new
+ stringstream, tmp_stream, and swap it with the argument output_model_stream
+ once it's ready, output_model_stream.swap(tmp_stream). Do not use
+ output_model_stream.clear(). It only clears out error state flag
+ (https://www.cplusplus.com/reference/ios/ios/clear/), while the content is the
+ same. It's cleaner to just declare a new one and swap.
+
+*/
+
+// The functions needed for backport model from v5 to v4.
+namespace {
+
+void writeArchiveV4(
+    PyTorchStreamWriter& writer,
+    const std::string& archive_name,
+    const c10::IValue& value) {
+  std::vector<char> data;
+
+  // Vector to capture the run-time class types during pickling the IValues
+  std::vector<c10::ClassTypePtr> memoizedClassTypes;
+  Pickler data_pickle(
+      [&](const char* buf, size_t size) {
+        data.insert(data.end(), buf, buf + size);
+      },
+      nullptr,
+      nullptr,
+      &memoizedClassTypes);
+  data_pickle.protocol();
+  data_pickle.pushIValue(value);
+  data_pickle.stop();
+  size_t i = 0;
+  std::string prefix = archive_name + "/";
+
+  for (const auto& td : data_pickle.tensorData()) {
+    WriteableTensorData writable_td = getWriteableTensorData(td);
+    std::string fname = prefix + c10::to_string(i++);
+    writer.writeRecord(fname, writable_td.data(), writable_td.sizeInBytes());
+  }
+  std::string fname = archive_name + ".pkl";
+  writer.writeRecord(fname, data.data(), data.size());
+}
+
+std::stringstream backport_v5_to_v4(std::stringstream& input_model_stream) {
```

Comment:
I should have being stricter about this, but please let's document each of this backport functions: what was the version bump about, and what the backport is accomplishing. It's less important about the code details, since per my comment above my suggestions would make the code more self documenting.